### DISCUSSION TITLE
How do I check if my machine supports floating point textures and how many floating point bits it supports?

### DISCUSSION THREAD

**Discussion by znw-test** [pixijs/pixijs#8817] [Q&A]
I wrote a program using PixiJS with floating point textures that worked fine on my computer, but I had an error when I moved it to another computer.

I created a floating point texture using PixiJS' RenderTexture class and tried to do something with the image on Mesh. Everything worked fine on my computer, but there were a lot of strange noises on the other one.
```js
RenderTexture.create({ width: originTexture.width, height: originTexture.height, type: TYPES.FLOAT })
```

When PixiJS is initialized, it prints information that says WebGL2 is used.

So I guess the cause of the problem may be the graphics card. 

So how do I use JavaScript to check if the graphics card supports floating point textures and how many floating point bits it supports?

Or if you have any other ideas, please let me know. Thank you!

PS: I try to set the texture format to gl.RGBA32F: `RenderTexture.create({ width: originTexture.width, height: originTexture.height, type: TYPES.FLOAT, format: gl.RGBA32F })`, but there is an error `WebGL: INVALID_ENUM: texImage2D: invalid format`.

**Comment by SuperSodaSea** [comment]
For WebGL 2 you can check these two WebGL extensions:
- [`EXT_color_buffer_half_float`](https://www.khronos.org/registry/webgl/extensions/EXT_color_buffer_half_float): `R16F`, `RG16F` and `RGBA16F` are supported.
    - Check it with `renderer.context.gl.getExtension('EXT_color_buffer_half_float')`
- [`EXT_color_buffer_float`](https://www.khronos.org/registry/webgl/extensions/EXT_color_buffer_float): `R16F`, `RG16F`, `RGBA16F`, `R32F`, `RG32F`, `RGBA32F` and `R11F_G11F_B10F` are supported.
    - Check it with `renderer.context.gl.getExtension('EXT_color_buffer_float')`
    - Or `renderer.context.extensions.colorBufferFloat`

Or you can just visit [https://webglreport.com/?v=2](https://webglreport.com/?v=2) and see if these two extensions exist for your browser.

**Reply by znw-test** [reply]
Thank you for your answer. I checked the WebGL extensions. I use the `gl.getSupportedExtensions()` and visit the website https://webglreport.com/?v=2. Both results show `EXT_color_buffer_half_float` and `EXT_color_buffer_float`. 

I found that the computer had made a mistake in calculating the formula `pow((cos(x * PI) + 1.0) * 0.5, 0.5)`. As x approaches 1, 3, and 5, some of the results are not as expected. And if I set the formula to `pow((cos(x * PI) + 1.0001) * 0.5, 0.5)`, everything seems normal.  I guess it is a precision problem, so how do I set `RenderTexture` to `RGBA32F`?

**Reply by SuperSodaSea** [reply]
Try `{ format: FORMATS.RGBA, type: TYPES.FLOAT }`?

**Reply by znw-test** [reply]
I didn't set the format manually. But I check the `texture.baseTexture.format` equals `FORMATS.RGBA`, both are `6408`.
Same as I use `RenderTexture.create({ width: originTexture.width, height: originTexture.height, type: TYPES.FLOAT, format: FORMATS.RGBA })`.
